framework/src/modules/random/utils.ts

```diff
@@ -60,7 +58,7 @@ export const getSeedRevealValidity = (
 
 	if (
 		!lastSeed ||
-		lastSeed.seedReveal.equals(cryptography.utils.hash(seedReveal).slice(0, SEED_REVEAL_HASH_SIZE))
+		lastSeed.seedReveal.equals(cryptography.utils.hash(seedReveal).slice(0, SEED_LENGTH))
```

Comment:
To your second point, break needs to remain otherwise there is a deepscan error: `Value assigned to variable 'maxheight' at this point is not used afterwards.`